skimage/color/colorconv.py

```diff
@@ -842,6 +842,9 @@ def gray2rgb(image, alpha=None):
             is_rgb = True
 
     if is_rgb:
+        warn('Non gray scale images conversion is now deprecated. In version '
+             '0.19, input image  will be considered as grey scale, whatever '
+             'is its shape.', FutureWarning, stacklevel=2)
```

Comment:
gray scale/grey scale -> grayscale, and remove "whatever is its shape". ie:

"Pass-through of possibly RGB images in `gray2rgb` is deprecated. In version 0.19, input arrays will always be considered grayscale, even if the last dimension has length 3 or 4. To prevent this warning and ensure compatibility with future versions, detect RGB images outside of this function."

We also need a way to silence this warning when this function is used correctly with a grayscale array with last axis shape (3,)...

Reply:
Do you mean that we need to add an argument (`nd` for example) to be set to True if the user wants effectively convert nD grayscale image to RGB?

Reply:
Yes, exactly. But `nd` isn't a great argument name, and anyway, in the future we would want to remove it. So I don't really know what to do here. =\